src/support/hash.h

```diff
@@ -29,10 +29,19 @@ template<typename T> inline std::size_t hash(const T& value) {
 
 // Combines two digests into the first digest. Use instead of `rehash` if
 // `otherDigest` is another digest and not a `size_t` value.
-static inline void hash_combine(std::size_t& digest, std::size_t otherDigest) {
-  // see boost/container_hash/hash.hpp
+static inline void hash_combine(std::uint32_t& digest,
+                                const std::uint32_t otherDigest) {
```

Comment:
I don't think `const` here is necessary. Also it seems this change force refactoring all call sites to use const cast